flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java

```diff
@@ -216,44 +217,74 @@ public void testGetCatalogs() throws Exception {
     @Test
     public void testGetSchemas() throws Exception {
         try (Connection connection = ENDPOINT_EXTENSION.getConnection()) {
-            connection.createStatement().execute("CREATE SCHEMA schema1");
-            connection.createStatement().execute("CREATE SCHEMA schema2");
-            connection.createStatement().execute("CREATE SCHEMA different");
-
+            initDefaultTestConnection(connection);
             // test all
-            java.sql.ResultSet resultAll = connection.getMetaData().getSchemas("hive", null);
-            assertSchemaEquals(
-                    ResolvedSchema.of(
-                            Column.physical("TABLE_SCHEMA", DataTypes.STRING()),
-                            Column.physical("TABLE_CAT", DataTypes.STRING())),
-                    resultAll.getMetaData());
+            testGetSchemasAll(connection);
+            // test schema pattern parameter
+            testGetSchemasWithPattern(connection);
+        }
+    }
 
-            List<List<String>> actual = new ArrayList<>();
-            while (resultAll.next()) {
-                actual.add(Arrays.asList(resultAll.getString(1), resultAll.getString(2)));
-            }
-            assertThat(new HashSet<>(actual))
-                    .isEqualTo(
-                            new HashSet<>(
-                                    Arrays.asList(
-                                            Arrays.asList("default", "hive"),
-                                            Arrays.asList("schema1", "hive"),
-                                            Arrays.asList("schema2", "hive"),
-                                            Arrays.asList("different", "hive"))));
+    private void testGetSchemasAll(Connection connection) throws Exception {
```

Comment:
I think we can resuse most of codes between `testGetSchemasAll` and `testGetSchemasWithPattern`